src/Expander/Task.hs

```diff
@@ -36,18 +36,18 @@ data ExpanderTask
   | AwaitingDefn Var Ident Binding SplitCorePtr Ty SplitCorePtr Syntax
     -- ^ Waiting on var, binding, and definiens, destination, syntax to expand
   | AwaitingType SplitTypePtr [AfterTypeTask]
-  | ExpandDecl DeclPtr Scope Syntax DeclValidityPtr
-    -- ^ Where to put it, the scope, the decl, where to put the phase
-  | ExpandMoreDecls ModBodyPtr Scope Syntax DeclValidityPtr
-    -- Depends on the binding of the name(s) from the decl and the phase
+  | ExpandDeclTree DeclTreePtr Scope Syntax DeclValidityPtr
+    -- ^ The tree node to fill, the scope, the decls, and the output environment.
+  | ExpandDependentDeclTree DeclTreePtr Syntax DeclValidityPtr
+    -- ^ The tree node to fill, the decls, and the input environment.
```

Comment:
I changed the comments again to describe the `DeclTreePtr`s as pointing to environments. Maybe `EnvPtr` would be a more acceptable name than `ScopeSetPtr`?

Reply:
We already have something called an `Env`, and it's not this.

More generally, I think it's important to keep the concept of environment (mapping core names to values), typing contexts (mapping core names to type schemes), and scope sets (which aren't really even conceptually mappings - the internal `Map` is used to make them have different natures at different times, but they're intended to be a bit more set-like).

What about `DeclOutputScopesPtr`?

Reply:
ScopeSets don't associate identifiers with definitions, they associate identifiers with their bindings, which might be definitions, but that also might be e.g. a local variable in a lambda.

Reply:
I have now updated the comments to talk about scopes, not environments.